File: src/main/resources/static/js/education.js
(async function () {
    const root = document.querySelector('#education-list');
    if (!root) return;
  
    try {
      const res = await fetch('/api/education', { headers: { 'Accept': 'application/json' } });
      if (!res.ok) throw new Error(`HTTP ${res.status}`);
      const schools = await res.json();
  
      root.innerHTML = schools.map(s => `
        <div class="rounded-2xl border p-4">
          <div class="flex items-baseline justify-between">
            <div>
              <div class="font-medium">${s.degree}</div>
              <div class="text-sm text-neutral-600">${s.school} — ${s.location}</div>
            </div>
            <div class="text-sm text-neutral-600">${s.dates}</div>
          </div>
          ${s.details ? `<p class="mt-2 text-sm text-neutral-700">${s.details}</p>` : ''}
        </div>
      `).join('');
    } catch (e) {
      console.error('Failed to load education', e);
      root.innerHTML = `<div class="text-sm text-neutral-500">Could not load education.</div>`;
    }
  })();
  

File: src/main/java/com/portfolio/demo/model/Education.java
package com.portfolio.demo.model;

public class Education {
    private String school;
    private String degree;
    private String location;
    private String dates;
    private String details;

    public Education() {}

    public Education(String school, String degree, String location, String dates, String details) {
        this.school = school;
        this.degree = degree;
        this.location = location;
        this.dates = dates;
        this.details = details;
    }

    public String getSchool() { return school; }
    public String getDegree() { return degree; }
    public String getLocation() { return location; }
    public String getDates() { return dates; }
    public String getDetails() { return details; }
}
